apps/mockingbird/README.md

```diff
@@ -0,0 +1,56 @@
+# Mockingbird - LON traffic generator
+
+Mockingbird is a Python script that leverages IAP/MQ protocol to create and control Internal LON devices (ILDs). The script can be utilized for quality assurance purposes of the SmartServer system. In further text, the internal LON devices are called **MBirds** in order to reduce the possibility of confusion with other similar abbreviations.
+
+MBirds have datapoints that reside on the SmartServer itself; in this case, we call this system Mockingbird Host (**MBH**). The MBirds' input datapoints are writable from the LonTalk network, and modifying the output datapoints produces data on the LonTalk network. MBirds can be created and provisioned, and their datapoints can be controlled as if they are actual physical devices. An example use case for this type of mechanism is to create MBirds, provision them on the unit under test (**UUT**), and generate load by modifying the MBirds' datapoints to observe and monitor the behavior under load of the unit under test.
+
+![](./docs/images/mockingbird_diagram.png)
+
+
+
+#### Prerequisites
+
+To create an internal device, you must first <u>import the corresponding XIF to the SmartServer using the Device Type widget, the CMS REST API or the DTP loader</u>.  Furthermore, when using IzoT Net Server with the SmartServer, the binary version of the resources needs to be added to your Lonworks/types folder.  It is also <u>required that the SmartServer be put into IMM mode</u> prior to instantiating internal devices using the script. In order to achieve external communication using the script, <u>MQTT ports (1883 and 8883) should be opened via the Features Configuration page</u> on the SmartServer web interface. Finally, the <u>SmartServer needs to be a Quad Core</u> (Serial number contains F or higher) and be running 3.64 or later.
```

Comment:
Regarding the dire warning about IMM - I see there is an undocumented switch --imm.  Can we just say instead that when doing device creation, that IMM is ensured to be set and not even bother the user about this?